### PR TITLE
Prepare TEE dataset download inside enclave

### PR DESCRIPTION
**Pull request by zguesmi**

### CODE REVIEW

File: src/main/java/com/iexec/common/utils/FileHelper.java [hunk 1]
```diff
@@ -119,47 +119,65 @@ public static File createFileWithContent(String filePath, byte[] data) {
...
-        } catch (IOException e) {
-            log.error("Failed to download file [fileUri:{}, exception:{}]",
-                    fileUri, e.getCause());
+        byte[] fileBytes = readFileBytesFromUrl(fileUrl);
+        if (fileBytes == null) {
+            log.error("Failed to download file [fileUrl:{}]", fileUrl);
+            return "";
+        }
+        boolean parentFolderAlreadyExisted = exists(parentFolderPath);
+        if (!createFolder(parentFolderPath)) {
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/utils/FileHelper.java]: `!parentFolderAlreadyExisted && !createFolder(parentFolderPath)` ?

File: src/main/java/com/iexec/common/utils/FileHelper.java [hunk 2]
```diff
@@ -119,47 +119,65 @@ public static File createFileWithContent(String filePath, byte[] data) {
...
             return "";
         }
+        log.info("Downloaded data [fileUrl:{}, filePath:{}]", fileUrl, filePath);
+        return filePath;
+    }
 
+    /**
+     * Read the content of the remote file located at the provided URI.
+     * @param url
+     * @return Optional describing an input stream with the file's content
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/utils/FileHelper.java]: optional ?

File: src/test/java/com/iexec/common/utils/FileHelperTests.java [hunk 3]
```diff
@@ -182,46 +189,94 @@ public void shouldZipFolder() {
...
+    public void shouldNotReadFileBytesFromBadUrl() {
+        byte[] bytes = FileHelper.readFileBytesFromUrl("http://bad-url");
+        assertThat(bytes).isNull();
+    }
+
+    // downloadFile(url, dir)
+
     @Test
     public void shouldDownloadFile() {
         //TODO 1 - Try https resources: https://iex.ec/wp-content/uploads/2018/12/token.svg
```
Comment (james-toussaint) [inline src/test/java/com/iexec/common/utils/FileHelperTests.java]: I think it works with SSL, no?

**Review by james-toussaint — APPROVED** [review]

File: src/main/java/com/iexec/common/utils/FileHelper.java (hunk 2 above)
Comment (zguesmi) [reply src/main/java/com/iexec/common/utils/FileHelper.java]: Old comment, my bad!

File: src/test/java/com/iexec/common/utils/FileHelperTests.java (hunk 3 above)
Comment (zguesmi) [reply src/test/java/com/iexec/common/utils/FileHelperTests.java]: :ok_hand:

**Pull request merged**